value-processor/src/org/immutables/value/processor/meta/NestedBuilderReflection.java

```diff
@@ -0,0 +1,222 @@
+      return true;
+    }
+
+    if (valueAttribute().typeKind().isList()) {
+      System.out.println(valueAttribute().firstTypeParameter());
+      TypeElement unboxedReturnType = valueAttribute()
+          .environment()
+          .processing()
+          .getElementUtils()
+          .getTypeElement(valueAttribute().firstTypeParameter());
+      return unboxedReturnType.getKind() == ElementKind.CLASS
+          || unboxedReturnType.getKind() == ElementKind.INTERFACE;
+    }
+
+    return false;
+  }
+
+  @Lazy
+  public boolean isImmutable() {
+    return unboxedReturnType().asElement().getAnnotation(Immutable.class) != null;
+  }
+
+  /**
+   * @return type mirror for the return type, or the template argument of a single collection
+   * element.
+   */
+  @Lazy
+  protected DeclaredType unboxedReturnType() {
+    if (valueAttribute().typeKind().isRegular()
+        && valueAttribute().returnType.getKind() == TypeKind.DECLARED) {
+      return (DeclaredType) valueAttribute().returnType;
+    } else if (valueAttribute().typeKind().isList()) {
+      TypeElement unboxedReturnType = valueAttribute()
+          .environment()
+          .processing()
+          .getElementUtils()
+          .getTypeElement(valueAttribute().firstTypeParameter());
+
+      if (unboxedReturnType.getKind() == ElementKind.CLASS
+          || unboxedReturnType.getKind() == ElementKind.INTERFACE) {
+        return (DeclaredType) unboxedReturnType.asType();
+      }
+    }
+
+    throw new RuntimeException(
+        String.format("Could not handle valueAttribuet %s", valueAttribute()));
+  }
+
+  @Lazy
+  protected Style.NestedBuilder immutablesNestedBuilderStyle() {
+    return new Style.NestedBuilder() {
+      @Override
+      public Class<? extends Annotation> annotationType() {
+        return Style.NestedBuilder.class;
+      }
+
+      @Override
+      public String getBuilderMethod() {
+        return getNestedImmutableValueType().constitution.style().builder();
+      }
+
+      @Override
+      public String buildMethod() {
+        return getNestedImmutableValueType().constitution.style().build();
+      }
+
+      @Override
+      public FromMethod fromMethod() {
+        return new FromMethod() {
+
+          @Override
+          public Class<? extends Annotation> annotationType() {
+            return FromMethod.class;
+          }
+
+          @Override
+          public Target target() {
+            return Target.BUILDER_INSTANCE;
+          }
+
+          @Override
+          public String methodName() {
+            return getNestedImmutableValueType().constitution.style().from();
+          }
+        };
+      }
+    };
+  }
+
+  @Lazy
+  public ValueType getNestedImmutableValueType() {
+    if (!isImmutable()) {
+      throw new RuntimeException(
+          "You should not be calling getNestedImmutableValueType for non immutable.");
+    }
+
+    ValueType nestedImmutable = new ValueType();
```

Comment:
Please see `ValueAttribute. initAttributeValueType()`. Basically, we can expand if condition to include `nestedBuilderApi()` (or better `nestedBuilderDetection/Discovery`)

```java
if ((containingType.constitution.style().deepImmutablesDetection()
    || containingType.constitution.style().nestedBuilderDetection())
     && containedTypeElement != null ....
```

once having `attributeValueType` initialized, we can then use to derive any builder models, the attribute value type should have its own style, not necessarily the one used by value type which contains the attribute.